src/devices/HardwareMonitor/OpenHardwareMonitorHttp.cs

```diff
@@ -0,0 +1,280 @@
+                        {
+                            var sensorToUpdate = _sensors.FirstOrDefault(x => x.Identifier == sensor.NodeId);
+                            if (sensorToUpdate != null)
+                            {
+                                sensorToUpdate.Value = sensor.Value;
+                            }
+                        }
+                    }
+
+                    _lastUpdateTime = DateTime.UtcNow;
+                }
+            }
+            catch (AggregateException ex)
+            {
+                _logger.LogWarning(ex, $"Unable to read /api/rootnode: {ex.Message}");
+                // If we failed to retrieve updates for some time, reset the fields.
+                if ((DateTime.UtcNow - _lastUpdateTime).Duration().TotalSeconds > UpdateInterval.TotalSeconds * 10) // (netstandard doesn't support operator* on TimeSpan)
+                {
+                    _hardware = new List<OpenHardwareMonitor.Hardware>();
+                    _sensors = new List<SensorHttp>();
+                }
+            }
+        }
+
+        private (List<OpenHardwareMonitor.Hardware> Hardware, List<SensorHttp> Sensors) CreateSensorTree(ComputerJson rootNode)
+        {
+            var newHardware = new List<OpenHardwareMonitor.Hardware>();
+            var newSensors = new List<SensorHttp>();
+
+            foreach (var hardware in rootNode.Hardware)
+            {
+                var hw = new OpenHardwareMonitor.Hardware(hardware.Name, hardware.NodeId, hardware.Parent, hardware.HardwareType);
+                newHardware.Add(hw);
+
+                foreach (var sensor in hardware.Sensors)
+                {
+                    SensorType sensorType;
+                    if (Enum.TryParse<SensorType>(sensor.Type, true, out sensorType))
+                    {
+                        var s = new SensorHttp(this, sensor.Name, sensor.NodeId, sensor.Parent, sensorType, sensor.Value);
+                        newSensors.Add(s);
+                    }
+                }
+            }
+
+            return (newHardware, newSensors);
+        }
+
+        public IList<OpenHardwareMonitor.Sensor> GetSensorList()
+        {
+            if (!_sensors.Any())
+            {
+                GetFullDataAndDecode(false);
+            }
+
+            return _sensors.Cast<OpenHardwareMonitor.Sensor>().ToList();
+        }
+
+        public IList<OpenHardwareMonitor.Hardware> GetHardwareComponents()
+        {
+            if (!_hardware.Any())
+            {
+                GetFullDataAndDecode(false);
+            }
+
+            return _hardware;
+        }
+
+        public void Dispose()
+        {
+            _version = string.Empty;
+            _httpClient.Dispose();
+        }
+
+        /// <summary>
+        /// Returns a non-rooted path combined from the input paths that uses forward slashes (for web apis)
+        /// </summary>
+        private string UnixPathJoin(string a, string b)
```

Comment:
Take a look at Uri methods to create the uri. Maybe TryCreate method.

Reply:
Doesn't work out-of-the-box, but with some trickery.